extensions/wikia/UserProfilePageV3/js/UserProfilePage.js

```diff
@@ -414,248 +401,263 @@ var UserProfilePage = {
 					UserProfilePage.bucky.flush();
 					window.location = UserProfilePage.reloadUrl;
 				}
-			},
-			error: UserProfilePage.error,
-			complete: function () {
+			}).fail(
+				UserProfilePage.error
+			).done(function () {
 				saveButton.prop('disabled', false);
+			});
+		},
+
+		/**
+		 * Hnadle error states after ajax requests
+		 * @param {string} [msg] Optional error message to display. Otherwise, default message will be used.
+		 */
+		error: function (msg) {
+			if (typeof msg !== 'string') {
+				msg = mw.message('oasis-generic-error');
 			}
-		});
-	},
-
-	/**
-	 * Hnadle error states after ajax requests
-	 * @param {string} [msg] Optional error message to display. Otherwise, default message will be used.
-	 */
-	error: function (msg) {
-		'use strict';
-
-		if (typeof msg !== 'string') {
-			msg = $.msg('oasis-generic-error');
-		}
-
-		UserProfilePage.bannerNotification.setContent(msg).show();
-	},
-
-	getFormData: function () {
-		'use strict';
-
-		var userData = {
-				name: null,
-				location: null,
-				occupation: null,
-				gender: null,
-				website: null,
-				twitter: null,
-				fbPage: null,
-				year: null,
-				month: null,
-				day: null
-			},
-			i,
-			userDataItem;
-
-		for (i in userData) {
-			userDataItem = $(document.userData[i]).val();
-			if (typeof (userDataItem) !== 'undefined') {
-				userData[i] = userDataItem;
-			}
-		}
 
-		userData.hideEditsWikis = document.userData.hideEditsWikis.checked ? 1 : 0;
-
-		return userData;
-	},
+			UserProfilePage.bannerNotification.setContent(msg).show();
+		},
+
+		getFormData: function () {
+			var userData = {
+					name: null,
+					location: null,
+					occupation: null,
+					gender: null,
+					website: null,
+					twitter: null,
+					fbPage: null,
+					year: null,
+					month: null,
+					day: null
+				},
+				i,
+				userDataItem;
 
-	refillBDayDaySelectbox: function (selectboxes) {
-		'use strict';
+			for (i in userData) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Restrict what this loop acts on by testing each property. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AForIn)